src/main/java/com/redhat/pantheon/servlet/AsciidocRenderingServlet.java

```diff
@@ -151,28 +147,30 @@ private Content generateHtml(SlingHttpServletRequest request, Resource resource)
         return c;
     }
 
-    private void cacheContent(SlingHttpServletRequest request, Resource resource, Content content) throws PersistenceException {
-        Map<String, Object> props = new HashMap<>();
-        props.put("jcr:data", content.html);
-        props.put("jcr:primaryType", "pant:moduleMetadata");
+    private void cacheContent(SlingHttpServletRequest request, Resource resource, Content content) {
+        try {
+            Node resourceNode = resource.adaptTo(Node.class);
+            Node cacheNode = resourceNode.getNode(CACHE_NODE_NAME);
```

Comment:
If you're referring to the original approach with the ModifiableValueMap, then I don't think there's much of a difference, honestly. I'm sure you already found the page at http://localhost:8080/system/console/adapters - the impression that I get is that depending on what the underlying object represents, then you may not have a choice whether you use Sling classes or JCR classes.

In this particular case, I don't remember why I needed to refactor the code to adapt to Node. I think I was originally trying to create the cache node there and somehow that made it easier - maybe because of the way I had to set the primaryType? - but regardless I'm obviously not doing that anymore. Still, I think this code is simple and direct, so that's why I kept it.